fintorch/datasets/invoice.py

```
@@ -0,0 +1,135 @@
+            os.makedirs(self.root, exist_ok=True)
+            os.makedirs(os.path.join(self.root, "raw"), exist_ok=True)
+            os.makedirs(os.path.join(self.root, "processed"), exist_ok=True)
+            os.makedirs(
+                os.path.join(self.root, "processed/training_data"), exist_ok=True
+            )
+            os.makedirs(
+                os.path.join(self.root, "processed/testing_data"), exist_ok=True
+            )
+        except OSError as e:
+            logging.error(f"Failed to create directories: {str(e)}")
+            raise RuntimeError(f"Failed to setup directories: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)